### ISSUE TITLE
error when incremental train with pretrained model

### ISSUE THREAD

**Issue by civex** [OpenNMT/OpenNMT-py#2032]
i tried to perform incremental train with another dataset and pretrained model, but the following error occurred.

----- error ------
/root/anaconda3/lib/python3.8/multiprocessing/resource_tracker.py:216: UserWarning: resource_tracker: There appear to be 6 leaked semaphore objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d '
Bus error (core dumped)
------------------

the pretrained model was trained on one GPU and incremental train is trained on two GPUs.

if GPU is set as one in the incremental train, there is no problem.

What is the problem?

**Comment by francoishernandez** [comment]
You need to give more details.
- version of OpenNMT-py used (both for pretraining and retraining)
- full command line(s)
- full logs (at least the full error trace)

**Comment by civex** [comment]
Thank you for checking the question!!

1. version
OpenNMT-py : 2.0.1
pytorch : 1.7.1

2. training config (yml)

src_vocab: data/train.vocab
tgt_vocab: data/train.vocab

src_vocab_size: 32002
tgt_vocab_size: 32002

vocab_size_multiple: 8
src_words_min_frequency: 5
tgt_words_min_frequency: 5
share_vocab: true
num_threads: 16

src_subword_model: model/spm.model
tgt_subword_model: model/spm.model
...
src_subword_type: sentencepiece
tgt_subword_type: sentencepiece

src_onmttok_kwargs: "{'mode': 'none', 'spacer_annotate': True}"
tgt_onmttok_kwargs: "{'mode': 'none', 'spacer_annotate': True}"
switchout_temperature: 1.0
tokendrop_temperature: 1.0
tokenmask_temperature: 1.0

src_seq_length: 512
tgt_seq_length: 512

data:
    corpus_1:
        path_src: data/train.src
        path_tgt: data/train.tgt
        transforms: [sentencepiece, filtertoolong]
    valid:
        path_src: data/valid.src
        path_tgt: data/valid.tgt
        transforms: [sentencepiece, filtertoolong]

log_file: train.log
log_file_level: DEBUG

**# for incremental training (retraining )
train_from: model/base_step_400000.pt**

save_model: model/incr
keep_checkpoint: 10
save_checkpoint_steps: 25000
average_decay: 0.0001
seed: 2345
report_every: 100
train_steps: 1000000
valid_steps: 25000
early_stopping: 5
early_stopping_criteria: [accuracy, ppl]

queue_size: 10000
bucket_size: 32768

**# for pretraining**
world_size: 1
gpu_ranks:
- 0

**# for incremental training (retraining )**
world_size: 2
gpu_ranks:
- 0
- 1
batch_type: "tokens"
batch_size: 4096
valid_batch_size: 1024
batch_size_multiple: 1
max_generator_batches: 0
accum_count: [4]
accum_steps: [0]

model_dtype: "fp32"
optim: "adam"
learning_rate: 2
warmup_steps: 10000
decay_method: "noam"
adam_beta1: 0.9
adam_beta2: 0.999
max_grad_norm: 0
label_smoothing: 0.1
param_init: 0
param_init_glorot: true
normalization: "tokens"

encoder_type: transformer
decoder_type: transformer
enc_layers: 12
dec_layers: 12
heads: 16
rnn_size: 512
word_vec_size: 512
position_encoding: true
transformer_ff: 2048
dropout_steps: [0]
dropout: [0.1]
attention_dropout: [0.1]
share_decoder_embeddings: true
share_embeddings: true


3. log
There is no other log except the error log in the question.
